bindings/sendgrid/sendgrid.go

```diff
@@ -0,0 +1,131 @@
+// ------------------------------------------------------------
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+// ------------------------------------------------------------
+
+package sendgrid
+
+import (
+	"errors"
+	"fmt"
+	"strconv"
+
+	"github.com/dapr/components-contrib/bindings"
+	"github.com/dapr/dapr/pkg/logger"
+	"github.com/sendgrid/sendgrid-go"
+	"github.com/sendgrid/sendgrid-go/helpers/mail"
+)
+
+// SendGrid allows sending of emails using the 3rd party SendGrid service
+type SendGrid struct {
+	metadata sendGridMetadata
+	logger   logger.Logger
+}
+
+type sendGridMetadata struct {
+	APIKey    string `json:"apiKey"`
+	EmailFrom string `json:"emailFrom"`
+	EmailTo   string `json:"emailTo"`
+	Subject   string `json:"subject"`
+}
+
+// NewSendGrid returns a new SendGrid bindings instance
+func NewSendGrid(logger logger.Logger) *SendGrid {
+	return &SendGrid{logger: logger}
+}
+
+// Helper to parse metadata
+func (sg *SendGrid) parseMetadata(meta bindings.Metadata) (sendGridMetadata, error) {
+	sgMeta := sendGridMetadata{}
+
+	// Required properties
+	if val, ok := meta.Properties["apiKey"]; ok && val != "" {
+		sgMeta.APIKey = val
+	} else {
+		return sgMeta, errors.New("SendGrid binding error: apiKey field is required in metadata")
+	}
+
+	// Optional properties
+	sgMeta.EmailFrom = meta.Properties["emailFrom"]
+	sgMeta.EmailTo = meta.Properties["emailTo"]
+	sgMeta.EmailFrom = meta.Properties["emailFrom"]
```

Comment:
this a dupe of line 49?

Reply:
Yeah dunno what happened there, copy/paste snafu